docs/SONATA_DEVELOPER_GUIDE.md

```diff
@@ -904,107 +904,98 @@ The indexing is rooted at a single group called indices which is a subgroup of a
 
-### <a name="neuron_config">Tying it all together - the network/circuit config file
+The datasets from the target_to_source group are defined symmetrically. From this symmetry is easy to infer that edges should be grouped by source, target pairs, otherwise an important overhead will be incurred in one or both of the indices.
 
-The config file is a .json file that defines the relative location of each part of the network:
-
-{
+### <a name="network_config">Tying it all together - the network/circuit config file
 
-    "target_simulator":"NEURON",
+The config file is a JSON file that defines the relative location of each part of the network.
 
-target_simulator can be "NEURON", “PyNN”, “NEST”, etc.  It specifies the intended target simulator of the circuit description.  For now, this field is intended as a declaration, and an implementation may decide to throw an error for unsupported targets.  In practice, mechanism and parameter names are tailored to the given target simulator.  For model_type=biophysical NEURON is supported, but not PyNN or NEST.
+It includes the following sections / values:
 
-The "manifest" section of the config file provides a convenient handle on setting variables that point to base paths.  These variables can be then used in the rest of the config file to point to various directories that share the first portion of the path.
+#### target_simulator
 
-    "manifest": {
+Can be "NEURON", "PyNN", "NEST", etc. It specifies the intended target simulator of the circuit description. For now, this field is intended as a declaration, and an implementation may decide to throw an error for unsupported targets. In practice, mechanism and parameter names are tailored to the given target simulator. For `model_type`=`biophysical` NEURON is supported, but not PyNN or NEST.
 
-        "$BASE_DIR": "/path/to/my/workspace",
-
-        "$NETWORK_DIR": "$BASE_DIR/networks",
-
-        "$COMPONENT_DIR": "$BASE_DIR/components"
-
-    },
+Example:
+```
+"target_simulator": "NEURON"
+```
 
-    "components_dir": {
+#### manifest
 
-The directory in which to find the neuronal morphology files:
+Provides a convenient handle on setting variables that point to base paths. These variables can be then used in the rest of the config file to point to various directories that share the first portion of the path. One can also use '.' in the beginning of any path to declare a path relevant to the network config file location.
 
-        "morphologies_dir": "$COMPONENT_DIR/morphologies",
+Example:
+```
+"manifest": {
+  "$BASE_DIR": "./",
+  "$NETWORK_DIR": "$BASE_DIR/networks",
+  "$COMPONENT_DIR": "$BASE_DIR/components"
+}
+```
 
-The directory in which to find the edge_types and point neuron node_types dynamics_params .json files, respectively:
+#### components_dir
 
-        "synaptic_models_dir": "$COMPONENT_DIR/synapse_dynamics",
+Base directories for different components, in particular:
```

Comment:
Since you modified this, I'd suggest to change all names to be singular: morphology_dir, synaptic_model_dir, mechanism_dir, ... because using plurals doesn't sound correct English to me (e.g. LD_LIBRARY_PATH is not LD_LIBRARIES_PATH). @kaeldai , do you agree?

Reply:
To me it sounds more natural to use plural. However you do have a point that there is much more precedence to use singular. I'd defer my vote to @CellAssembly since as a modeler he's the one who would be looking at it the most.

Reply:
Thanks @kaeldai. I do agree that the plural version is more natural. It is a directory of "morphologies" and not of a single "morphology". One can also say directory of "morphology files" but then the plural is in there also in the word "files" and so to me plural is more natural. But in all honesty, this is a minor point and I am okay if singular is chosen.

Reply:
Let's keep the names as they are then.